backend/pkg/database/database.go

```diff
@@ -35,24 +35,44 @@ const (
 	DBDriverTypeLocal    = "LOCAL"
 )
 
+// The following views can contain lots of rows and if they are simple views it will cause a lot of slow queries.
+// In the case of materialized views slow queries can happen when refreshing the views,
+// but reading these views will not cause slow queries when navigating on the UI.
+// Materialized view is only supported by PostgreSQL.
+var materializedViews = []string{
+	applicationViewName,
+	resourceViewName,
+	packageViewName,
+	vulnerabilityViewName,
+}
+
 // order is important, need to drop views in a reverse order of the creation.
 var viewsList = []string{
 	"cis_d_b_checks_view",
-	"vulnerabilities_view",
+	vulnerabilityViewName, // MATERIALIZED in case of postgres.
 	"package_resources_info_view",
-	"packages_view",
+	packageViewName, // MATERIALIZED in case of postgres.
 	"package_severities",
-	"resources_view",
+	resourceViewName, // MATERIALIZED in case of postgres.
 	"resource_cis_d_b_checks_view",
 	"resource_severities",
 	"new_vulnerabilities_view",
-	"applications_view",
+	applicationViewName, // MATERIALIZED in case of postgres.
 	"application_cis_d_b_checks",
 	"application_severities",
 	"ids_view",
 	"licenses_view",
 }
 
+const (
+	createViewCommand              = "CREATE VIEW"
+	createMaterializedViewCommand  = "CREATE MATERIALIZED VIEW"
+	dropViewCommand                = "DROP VIEW IF EXISTS %s"
+	dropMaterializedViewCommand    = "DROP MATERIALIZED VIEW IF EXISTS %s;"
+	refreshMaterializedViewCommand = "REFRESH MATERIALIZED VIEW CONCURRENTLY %s;"
+	initMaterializedViewCommand    = "REFRESH MATERIALIZED VIEW %s;"
```

Comment:
init? or refresh?

Reply:
We can refresh materialized views concurrently, but it isn't working without populated data, so this is the reason why I used `init` in the const name.

Reply:
I think it is better to call it refreshMaterializedViewCommand and refreshMaterializedViewConcurrentlyCommand just for the clarity of things... what ever you think